File: python/test_fizzbuzz.py
import pytest
from fizzbuzz import fizzbuzz


def test_with_zero():
    assert fizzbuzz(0) == 0

def test_same_number_with_1_2_and_4():
    assert fizzbuzz(1) == 1
    assert fizzbuzz(2) == 2
    assert fizzbuzz(4) == 4

def test_print_Fizz_when_multiple_of_3_only():
    assert fizzbuzz(3) == 'Fizz'
    assert fizzbuzz(15) != 'Fizz'
    assert fizzbuzz(9) == 'Fizz'
    assert fizzbuzz(21) == 'Fizz'
    assert fizzbuzz(30) != 'Fizz'
    assert fizzbuzz(26) != 'Fizz'

def test_print_Fizz_when_multiple_of_5_only():
    assert fizzbuzz(5) == 'Buzz'
    assert fizzbuzz(15) != 'Buzz'
    assert fizzbuzz(25) == 'Buzz'
    assert fizzbuzz (30) != 'Buzz'
    assert fizzbuzz(6) != 'Buzz'
    assert fizzbuzz(18) != 'Buzz'

def test_print_Fizz_when_multiple_of_3_and_5():
    assert fizzbuzz(15) == 'FizzBuzz'
    assert fizzbuzz(45) == 'FizzBuzz'
    assert fizzbuzz(20) != 'FizzBuzz'
    assert fizzbuzz(50) != 'FizzBuzz'

File: ruby/test-fizzbuzz.rb
require_relative 'fizzbuzz'
require 'minitest/autorun'

class TestFizzBuzz < Minitest::Test

  def testWithZero
    assert_equal fizzbuzz(0), 0
  end

  def test_same_number_with_one_two_and_four
    assert_equal fizzbuzz(1), 1
    assert_equal fizzbuzz(2), 2
    assert_equal fizzbuzz(4), 4
  end

  def test_print_fizz_when_multiple_of_three_only
    assert_equal fizzbuzz(3), 'Fizz'
    refute_equal fizzbuzz(15), 'Fizz'
    assert_equal fizzbuzz(9), 'Fizz'
    assert_equal fizzbuzz(21), 'Fizz'
    refute_equal fizzbuzz(30), 'Fizz'
    refute_equal fizzbuzz(26), 'Fizz'
  end

  def test_print_fizz_when_multiple_of_5_only
    assert_equal fizzbuzz(5), 'Buzz'
    refute_equal fizzbuzz(15), 'Buzz'
    assert_equal fizzbuzz(25), 'Buzz'
    refute_equal fizzbuzz(30), 'Buzz'
    refute_equal fizzbuzz(6), 'Buzz'
    refute_equal fizzbuzz(18), 'Buzz'
  end

  def test_print_fizz_when_multiple_of_3_and_5
    assert_equal fizzbuzz(15), 'FizzBuzz'
    assert_equal fizzbuzz(45), 'FizzBuzz'
    refute_equal fizzbuzz(20), 'FizzBuzz'
    refute_equal fizzbuzz(50), 'FizzBuzz'
  end
end

